src/configuration.ts

```diff
@@ -41,6 +41,36 @@ function fixConfig(ts: TSCommon, config: _ts.ParsedCommandLine) {
   return config;
 }
 
+function readTsConfigExtendForTsNodeRecursively(
+  cwd: string,
+  ts: TSCommon,
+  rawConfig: { [key: string]: any } = {}
+): TsConfigOptions {
+  const configKey = 'ts-node';
+  const recurse = (
+    extending: { [key: string]: any } = {},
+    original: { [key: string]: any } = {}
+  ) => {
+    return readTsConfigExtendForTsNodeRecursively(cwd, ts, {
+      [configKey]: { ...extending[configKey], ...original[configKey] },
+    });
+  };
+
+  const { extends: extendFile, ...config } = rawConfig;
+
+  const result =
+    extendFile === undefined
+      ? filterRecognizedTsConfigTsNodeOptions(config[configKey])
+      : [extendFile]
+          .filter((x) => x && typeof x === 'string')
+          .map((x) => x as string)
+          .map((extended) => resolve(cwd, extended))
```

Comment:
Does this work when ./bar/tsconfig.json refers to ./bar/tsconfig2.json by relative path? Because the path is not relative to cwd?  Does it work for `"extends": "@tsconfig/node14/tsconfig.json"`?  (https://github.com/tsconfig/bases)

Reply:
Regarding relative paths... 
Not sure if I want to say yes, I do use this myself and I my cwd wasn't relative, but then again I am doing other stuff that might impact it. 
Good call, would do best to extract `dirname/basename` from current config in order to make relative paths work - but I gotta check first. 
There is another approach and that is to let `ts` itself try to find the file, there is a function that for that last I checked. 


Regarding `@tsconfig/node14/tsconfig.json`
Yeah no chance that would work. 
I am not familiar with that, is that something you distribute or TS? 
If it's the latter, then I'd reckon there wouldn't be a `ts-node` key there?

Reply:
We'll need tests for this, and we'll need to support typescript's `"extends"` resolution behavior.  @tsconfig/node14 was just an example; our users can choose to `"extends"` from anywhere that typescript supports, and we'll need to match that behavior.  Ideally, we figure out how TypeScript is doing this logic and do the exact same, using as many TypeScript APIs as possible to avoid duplication.

https://www.typescriptlang.org/tsconfig#extends

Reply:
Right, if TS exposes that, then I guess best bet is to use it, but to reimplement it is overkill. 
In general, I think the simplest approach is just following the links.

In regards to your previous points about relative paths, this is an easy fix: 
```ts
.map((extended) => path.isAbsolute() ?  [path] : [currentConfigPath, extendPath]
.map((extended) => resolve(currentConfigPath, extended))
```

However I don't care much for `extends: "@tsconfig/node14/tsconfig.json"`, on the premise that I just don't approve of this path they're going down on.

Reply:
Using relative paths and not CWD now.

Reply:
> However I don't care much for extends: "@tsconfig/node14/tsconfig.json", on the premise that I just don't approve of this path they're going down on.

Unfortunately tsc already does this so we need to match.  Otherwise it gets confusing and we need to document how our behavior deviates, and I'll be the one answering the issues that get files.

I certainly understand why you might choose not to use this feature on your projects, but for `ts-node`, we need to match TSC's behavior or else we're making things confusing.